var/spack/repos/builtin/packages/boost/python_jam.patch

```diff
@@ -0,0 +1,42 @@
+diff --git a/tools/build/src/tools/python.jam b/tools/build/src/tools/python.jam
+index 90377ea..123f66a 100644
+--- a/tools/build/src/tools/python.jam
++++ b/tools/build/src/tools/python.jam
+@@ -493,6 +493,10 @@ local rule probe ( python-cmd )
+                 sys.$(s) = [ SUBST $(output) \\<$(s)=([^$(nl)]+) $1 ] ;
+             }
+         }
++         # Try to get python abiflags
++        full-cmd = $(python-cmd)" -c \"from sys import abiflags; print(abiflags, end='')\"" ;
```

Comment:
Confirmed that the patch doesn't work for Python 2. I decided it would be easier/safer to only apply the patch when we are building with Python 3. Now both the Python 2 and 3 builds of Boost work for me!